test/core/client_channel/lb_policy/pick_first_test.cc

```diff
@@ -66,15 +129,28 @@ TEST_F(PickFirstTest, Basic) {
   ASSERT_NE(picker, nullptr);
   // Picker should return the same subchannel repeatedly.
   for (size_t i = 0; i < 3; ++i) {
-    EXPECT_EQ(ExpectPickComplete(picker.get()), kAddressUri);
+    EXPECT_EQ(ExpectPickComplete(picker.get()), kAddressUris[0]);
   }
 }
 
+TEST_F(PickFirstTest, WithShuffle) {
+  constexpr size_t kPolicyUpdates = 20;
+  constexpr std::array<absl::string_view, 6> kAddressUris = {
+      "ipv4:127.0.0.1:443", "ipv4:127.0.0.1:444", "ipv4:127.0.0.1:445",
+      "ipv4:127.0.0.1:446", "ipv4:127.0.0.1:447", "ipv4:127.0.0.1:448"};
+  // With shuffling, different addresses should be returned.
+  EXPECT_GT(ResetPickedAddress(kAddressUris, kPolicyUpdates, true).size(), 1);
+  // Without shuffling, the same address will always be returned
+  EXPECT_EQ(ResetPickedAddress(kAddressUris, kPolicyUpdates, false).size(), 1);
+}
+
 }  // namespace
 }  // namespace testing
 }  // namespace grpc_core
 
 int main(int argc, char** argv) {
+  grpc_core::testing::ScopedExperimentalEnvVar env_var(
```

Comment:
I think we should set this inside of the `WithShuffle` test rather than doing it globally.  In particular, note that we need a test that shows that if the env var is not set, then the pick_first policy does not actually look at the config field, and it always tries the addresses in the order in which they were provided to it.